### DISCUSSION TITLE
Blog 1: Unser Team und Vision

### DISCUSSION THREAD

**Discussion by Max-Rohrhirsch** [DHBW-UMOC/UMOC#1] [Announcements]
# Willkommen bei UMOC: Der Messangerdienst der nächsten Generation!

In einer Welt, in der Kommunikation schnell und unkompliziert sein sollte, tritt UMOC auf den Plan - unser neuer Messangerdienst, der nicht nur die grundlegenden Funktionen eines klassischen Messengers bietet, sondern auch eine Reihe von optionalen Features, die Benutzererfahrung und Interaktion auf ein ganz neues Level heben. UMOC haucht der online Kommunikation, durch spaßige Features Leben ein. Entwickelt mit den modernen Technologien SpringBoot und Angular, setzen wir auf eine benutzerfreundliche Oberfläche und leistungsstarke Backend-Funktionalitäten, um Ihnen eine überlegene Kommunikationserfahrung zu bieten.

## Das Team

- **Max Rohrhirsch** _[PO & Entwickler]_
- **Lucas Mohr** _[Scrum Master & Entwickler]_
- **Pascal Lingenfelser** _[Entwickler]_
- **Jan Mathis Schmid** _[Entwickler]_
- **Adrian Schön** _[Entwickler]_

## Tech-Stack & Aufbau

- Der **Server** ist ein Python Projekt mit Flask, welches eine API erstellt und eine postgres Datenbank nutzt.
- Das **Web Frontend** ist ein Angular Projekt und ist das GUI welches Leute nutzten. Im Hintergrund greift es auf die Flaks API zu. Das Web Frontend ist somit der Client.

## Warum UMOC?

Der Messangermarkt ist überflutet mit Anbietern, die oft nur die gleichen üblichen Funktionen anbieten. UMOC hebt sich jedoch ab und bietet erweiterte Features, die andere Anbieter teuer verkaufen. Hier sind einige der einzigartigen Funktionen, auf die wir stolz sind:

### Einblick in Passwörter

Haben Sie schon mal versucht einen Account zu erstellen, doch ihr Passwort war vergeben? Nein? Dann wird sich dies mit UMOC ändern. UMOC bietet die Möglichkeit in echtzeit einzusehen,welche Nutzer dieses Passwort bereits verwenden.

### Live-Typing

Keine Geheimnisse mehr! Eines der aufregendsten Features von UMOC ist die Möglichkeit, zu sehen, was ein Benutzer eintippt, bevor er es absendet. Dies fördert nicht nur eine nahtlose Kommunikation, sondern könnte auch zu mehr Verständnis führen, wenn Missverständnisse auftreten. 

### Gastkonto-Funktion

Mit unseren Gastkonten können Benutzer einfach und schnell ohne Registrierung am Geschehen teilnehmen. Dies ist besonders nützlich für spontane Gruppenunterhaltungen oder Events.

### Standortangehängte Nachrichten

Jede Nachricht, die Sie senden, enthält automatisch Ihren Standort. Das macht es einfacher, sich zu koordinieren, wenn Sie mit Freunden oder Kollegen unterwegs sind.

### Innovative Unterhaltungsfunktionen

UMOC geht sogar noch einen Schritt weiter:
- **Gelöschte Nachrichten** werden zu ihrer Planungssicherheit gespeichert, sodass jeder sehen kann, welche Nachrichten gelöscht wurden.
- **Blockierte Nutzer** dürfen eine letzte Nachricht senden.
- **VoteToKick** und **VoteToTimeout** in Gruppen sorgen dafür, dass Störenfriede, demokratisch, von Diskussionen ausgeschlossen werden können.

### Der Betrunkenen Modus

Und das Beste? Der "Betrunkenen Modus"! Wenn Sie etwas Spaß wollen, können Sie aktivieren, dass Ihre Nachrichten absichtlich falsch geschrieben werden. Lustige Interaktionen und Lacher sind garantiert!

## Alles ist optional

Eine der besten Eigenschaften von UMOC ist die Anpassungsfähigkeit. Alle diese Funktionen sind optional und können je nach Nutzerwunsch aktiviert oder deaktiviert werden. Wir glauben daran, dass jeder das Erlebnis anpassen sollte, das am besten zu ihm passt.

## Auf einen neuen Kommunikationsweg

UMOC ist mehr als nur ein Messangerdienst; es ist der Beginn einer neuen Ära in der digitalen Kommunikation. Unsere ambitionierten Features bieten nicht nur Unterhaltung, sondern auch hilfreiche Tools, um Ihre Kommunikation in verschiedenen sozialen Szenarien zu verbessern und den meisten Spaß herauszuholen.

Bleiben Sie dran für weitere Updates, Tutorials und spannende Ankündigungen. Unser Team arbeitet hart an der Verfeinerung und Verbesserung der Plattform, und wir können es kaum erwarten, Ihnen das vollständige Erlebnis von UMOC vorzustellen!

Lassen Sie uns die Zukunft der Kommunikation gemeinsam gestalten! Melden Sie sich noch heute bei UMOC an und haben Sie Spaß beim Chatten!

**Comment by TimHitschfeld** [comment]
UMOC bietet interessante und innovative Features, die sich deutlich von anderen Messengern abheben. Einige Funktionen, wie das Einsehen von Passwörtern und die automatische Standortfreigabe, werfen jedoch Sicherheits- und Datenschutzbedenken auf. Es wäre wichtig, mehr Kontrolle für die Nutzer einzubauen, um die Privatsphäre zu schützen. Besonders positiv ist die Flexibilität, Funktionen optional anzubieten, was ein personalisiertes Erlebnis ermöglicht. Insgesamt ein spannendes Konzept, das noch Feinschliff bei sicherheitsrelevanten Aspekten braucht.

Euer TripAhead Team
/TH

**Reply by schmid-di** [reply]
Vielen Dank für das Feedback :)
Sämtliche Features sind im Auslieferungszustand deaktiviert, bis auf unsere Verschlüsselung. Natürlich ist es unsere höchste Priorität unsere Nutzer weitesgehend zu schützen, sofern sie es denn wollen. Im Bezug auf deine Sicherheits- und Datenschutzbedenken, kann ich dich beruhigen, wir sind der Meinung, dass weniger Kontrolle über die Nutzer mehr ist.

Viele Grüße und vielen Dank
Euer UMOC Team
/JMS /PL

**Comment by JustMischka** [comment]
UMOC hat wirklich kreative Ideen, besonders mit dem "Betrunkenen Modus" – das bringt definitiv frischen Wind in den Messenger-Markt. Trotzdem sehe ich bei Funktionen wie "Einblick in Passwörter" und "Live-Typing" ein paar Herausforderungen in Sachen Datenschutz und Sicherheit. Da wäre es super, wenn ihr da mehr Infos gebt, um Bedenken auszuräumen. Alles in allem habt ihr einen vielversprechenden Start hingelegt, der mit ein paar Anpassungen richtig groß rauskommen könnte. Gute Arbeit!

Viele Grüße 
Team Jump and Run (MG)
